src/components/Divider.tsx

```diff
@@ -1,70 +1,172 @@
-import React, { useEffect } from 'react'
-import { useDispatch } from 'react-redux'
-import { useSelector } from 'react-redux'
+import { useEffect, useRef, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
 import { css } from '../../styled-system/css'
 import { SystemStyleObject } from '../../styled-system/types'
 import Path from '../@types/Path'
+import State from '../@types/State'
+import ThoughtId from '../@types/ThoughtId'
 import { setCursorActionCreator as setCursor } from '../actions/setCursor'
-import { DIVIDER_MIN_WIDTH, DIVIDER_PLUS_PX } from '../constants'
+import { DIVIDER_MIN_WIDTH } from '../constants'
 import attributeEquals from '../selectors/attributeEquals'
+import { getAllChildrenAsThoughts } from '../selectors/getChildren'
 import rootedParentOf from '../selectors/rootedParentOf'
+import editingValueUntrimmedStore from '../stores/editingValueUntrimmed'
 import fastClick from '../util/fastClick'
 import head from '../util/head'
+import isDivider from '../util/isDivider'
+
+/** Custom hook to fetch divider-related data from the state. */
+const useDividerData = (path: Path) => {
+  return useSelector(
+    (state: State) => {
+      const parentPath = rootedParentOf(state, path)
+      const parentId = head(parentPath)
+      const grandParentPath = parentId ? rootedParentOf(state, parentPath) : []
+      const grandParentId = head(grandParentPath)
+      const children = parentId ? getAllChildrenAsThoughts(state, parentId) : []
+      const childrenWithoutDividers = children.filter(child => !isDivider(child.value))
+      const isOnlyChild = childrenWithoutDividers.length === 0
+      const isTableView =
+        (parentId && attributeEquals(state, parentId, '=view', 'Table')) ||
+        (grandParentId && attributeEquals(state, grandParentId, '=view', 'Table'))
+
+      let thoughtsAtSameDepth: { id: ThoughtId; value: string }[] = []
+
+      if (isTableView && isOnlyChild && grandParentId) {
+        const parentSiblings = getAllChildrenAsThoughts(state, grandParentId).filter(child => !isDivider(child.value))
+        thoughtsAtSameDepth = parentSiblings.flatMap(parent => {
+          const childrenOfParent = parent.id ? getAllChildrenAsThoughts(state, parent.id) : []
+          return childrenOfParent.filter(child => !isDivider(child.value))
+        })
+      }
+
+      return {
+        isOnlyChild,
+        isTableView,
+        children,
+        thoughtsAtSameDepth,
+      }
+    },
+    // Ensure the selector runs only when values change, excluding ref changes to avoid redundant re-renders.
+    (prev, next) => JSON.stringify(prev) === JSON.stringify(next),
+  )
+}
+
+let canvas: HTMLCanvasElement | null = null
+let context: CanvasRenderingContext2D | null = null
+
+/**
+ * Initializes the canvas and context for text width calculation.
+ */
+const initializeCanvas = () => {
+  if (!canvas) {
+    canvas = document.createElement('canvas')
+    context = canvas.getContext('2d')
+  }
+}
```

Comment:
For readability, I would prefer to simply return a new context for each function call:

```suggestion
/**
 * Initializes the canvas and context for text width calculation.
 */
const getCanvasContext = () => {
  const canvas = document.createElement('canvas')
  return canvas.getContext('2d')
}
```

However, if we want to cache the canvas/context for performance reasons*, we can still simplify this by removing the `initializeContext` function altogether.

\* Running some really rough benchmarks on my machine, it seems to take about 0.01ms to create a canvas context and about 0.005ms to measure the text. This isn't _super_ concerning, but this function could be executed many times in parallel for large thoughtspaces with many dividers. For that reason, I agree with your choice to cache the canvas context (i.e. the second suggestion in this comment).

```suggestion
/** Initialize a canvas context for text width calculation */
let canvas = document.createElement('canvas')
let context = canvas.getContext('2d')
```